agent/ui_endpoint.go

```diff
@@ -813,8 +813,19 @@ func (s *HTTPHandlers) UIMetricsProxy(resp http.ResponseWriter, req *http.Reques
 	// Replace prefix in the path
 	subPath := strings.TrimPrefix(req.URL.Path, "/v1/internal/ui/metrics-proxy")
 
-	// Append that to the BaseURL (which might contain a path prefix component)
-	newURL := cfg.BaseURL + subPath
+	// This prevents path traversal while preserving URL structure
+	cleanedSubPath := path.Clean(subPath)
+
+	// Parse the base URL to get its components
+	baseURL, err := url.Parse(cfg.BaseURL)
```

Comment:
Can the `cfg.BaseURL` here also be cleaned? It also seems like it's used in other places like in logs at line 868. We could benefit from using the cleaned value here.